ygnmi/ygnmi.go

```diff
@@ -303,6 +303,22 @@ func (w *Watcher[T]) Await() (*Value[T], error) {
 	return w.lastVal, err
 }
 
+// AwaitWithContext waits for the watch to finish in the same way as Await, but will
+// return immediately if the context supplied is completed.
+func (w *Watcher[T]) AwaitWithContext(ctx context.Context) (*Value[T], error) {
```

Comment:
is canceling the context passed to Watch not sufficient? that context is passed to gRPC so it should cancel the subscribe and the watch

Reply:
i think the "right" way to handle that would be the propagate the context to the predicate, so it could cancel long running operating gracefully. Now, that would be backwards incompatible change.

Waiting for the pred to finish might actually be desirable because if its blocking indefinitely it's noticeable. (not silently leaking a goroutine)

i think having 2 contexts that cancel different things will be source of confusion (and/or bugs)

Reply:
The user can choose if we allow `Watch` and `pred` to both take `context.Context`. At that point, then the user can give a single context, if they are OK with it being cancelled, or two if they want to control this separately.

The use case here is that I have code that creating watchers that needs to also gracefully shut down, and can't indefinitely hang in that scenario - so I'd like to be able to have a way to cancel a context and get it to return, ideally also not leaking goroutines. :-) 

I can't quite tell from the comment whether you're against adding a `context.Context` to `pred`?

Reply:
Resolved, see note for future PR archaelogists.